util/topgen/intermodule.py

```diff
@@ -0,0 +1,111 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+from typing import Dict
+
+from .lib import *
+
+
+def elab_intermodule(topcfg):
+    """Check the connection of inter-module and categorize them
+
+    In the top template, it uses updated inter_module fields to create
+    connections between the modules (incl. memories). This function is to
+    create and check the validity of the connections `inter_module` using IPs'
+    `inter_signal_list`.
+    """
+
+    list_of_intersignals = []
+
+    if "inter_signal" not in topcfg:
+        topcfg["inter_signal"] = {}
+
+    # Gather the inter_signal_list
+    instances = topcfg["module"] + topcfg["memory"]
+
+    intermodule_instances = [x for x in instances if "inter_signal_list" in x]
+
+    for x in intermodule_instances:
+        for sig in x["inter_signal_list"]:
+            # Add instance name to the entry and add to list_of_intersignals
+            sig["inst_name"] = x["name"]
+            list_of_intersignals.append(sig)
```

Comment:
i'm reading this top to bottom, so pardon if this is explained later.
How does this work for modules that are instantiated multiple times...?

Reply:
o nvm.